src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -1613,6 +1613,26 @@ til::color Terminal::GetColorForMark(const Microsoft::Console::VirtualTerminal::
     }
 }
 
+std::wstring_view Terminal::CurrentCommand() const
+{
+    if (_currentPromptState != PromptState::Command ||
+        _scrollMarks.size() == 0)
+    {
+        return L"";
+    }
+
+    const auto& curr{ _scrollMarks.back() };
+    const auto& start{ curr.end };
+    const auto& end{ _activeBuffer().GetCursor().GetPosition() };
+
+    const auto line = start.y;
+    const auto& row = _activeBuffer().GetRowByOffset(line);
+    const auto rowText = row.GetText();
+    const auto commandText = rowText.substr(start.x, end.x-start.x);

```

Comment:
Hmm... This won't work. `start` and `end` are in column, but the string isn't.
I was always intending to add a `columnBegin` and `columnEnd` parameter to `GetText()` which default to `0` and "row width" respectively. That way you could safely slice out a piece of text like this. If you don't feel comfortable making such a change to `ROW`, let me know and we can collaborate on this!

Reply:
I pushed a commit which adds such a method. I haven't tested it though.